pkg/services/oauthtoken/oauth_token_test.go

```diff
@@ -270,3 +273,206 @@ func (f *FakeAuthInfoStore) UpdateAuthInfo(ctx context.Context, cmd *login.Updat
 func (f *FakeAuthInfoStore) DeleteAuthInfo(ctx context.Context, cmd *login.DeleteAuthInfoCommand) error {
 	return f.ExpectedError
 }
+
+func TestService_TryTokenRefresh(t *testing.T) {
+	type testCase struct {
+		desc      string
+		identity  *authn.Identity
+		oauthInfo *social.OAuthInfo
+		setupEnv  func(cache *localcache.CacheService, authInfoService *authinfotest.FakeService)
+
+		expectedErr error
+	}
+
+	tests := []testCase{
```

Comment:
Missing a case:
* with a valid token that does not need refresh.
* access token is expired => refresh 
* id token is expired => refresh

Reply:
Other missing cases:
- should skip and set the cache when the token doesn't need to be refreshed